content/ansible-tower-advanced/3-awx-collection-intro/_index.md

```diff
@@ -0,0 +1,281 @@
+  become: false
+  gather_facts: false
+  tasks:
+  - name: Create an inventory
+    awx.awx.tower_inventory:
+      name: AWX inventory
+      organization: Default
+  - name: Add hosts to inventory
+    awx.awx.tower_host:
+      name: "{{  item }}"
+      inventory: AWX inventory
+      state: present
+    loop:
+      - {{< param "internal_host1" >}}
+      - {{< param "internal_host2" >}}
+  - name: Machine Credentials
+    awx.awx.tower_credential:
+      name: AWX Credentials
+      kind: ssh
+      organization: Default
+      inputs:
+        username: ec2-user
+        ssh_key_data: "{{ lookup('file', '~/.ssh/aws-private.pem' ) }}"
+```
+
+</p>
+<hr/>
+</details>
+
+{{% notice info %}}
+If you run this Ansible Playbook multiple times, you will notice the **awx.awx.tower_credential** module is not idempotent! Since we store the SSH key encrypted, the Ansible Module is unable to verify it has already been set and didn't change. This is what we want and expect from a secure system, but it also means Ansible has no means to verify it and hence overrides the password every time the Ansible Playbook is executed.
+{{% /notice %}}
+
+## Create Projects
+
+The Ansible content used in this lab is hosted on Github. The next step is to add a project to import the Ansible Playbooks. As before, try to figure out the necessary parameters by reading the documentation of the **awx.awx.tower_project** module documentation.
+
+<details><summary><b>Click here for Solution</b></summary>
+<hr/>
+<p>
+
+```yaml
+---
+- name: Configure automation controller
+  hosts: localhost
+  become: false
+  gather_facts: false
+  tasks:
+  - name: Create an inventory
+    awx.awx.tower_inventory:
+      name: AWX inventory
+      organization: Default
+  - name: Add hosts to inventory
+    awx.awx.tower_host:
+      name: "{{  item }}"
+      inventory: AWX inventory
+      state: present
+    loop:
+      - {{< param "internal_host1" >}}
+      - {{< param "internal_host2" >}}
+  - name: Machine Credentials
+    awx.awx.tower_credential:
+      name: AWX Credentials
+      kind: ssh
+      organization: Default
+      inputs:
+        username: ec2-user
+        ssh_key_data: "{{ lookup('file', '~/.ssh/aws-private.pem' ) }}"
+  - name: AWX Project
+    awx.awx.tower_project:
+      name: AWX Project
+      organization: Default
+      state: present
+      scm_update_on_launch: True
+      scm_delete_on_update: True
+      scm_type: git
+      scm_url: https://github.com/goetzrieger/ansible-labs-playbooks.git
+```
+
+</p>
+<hr/>
+</details>
+
+If you wonder why **awx.awx.tower_credential** is not idempotent, read the info box in the [previous chapter](#create-machine-credentials).
+
+## Create Job Templates
+
+Before running an Ansible **Job** from your Automation Controller cluster you must create a **Job Template**, again business as usual for Automation Controller users. For this part of the Ansible Playbook, we will use the **awx.awx.tower_job_template** module.
+
```

Comment:
Again, they should try to write the Playbook w/o the solution, but you need to give them the parameters needed.

Reply:
fixed